src/hcf-release/jobs/uaa-create-user/templates/run.erb

```diff
@@ -150,30 +115,18 @@ else
                         %> --autoapprove '<%= client_data['autoapprove'] %>' <%
                     end
                 end
-            %>)"
-
-          # Test to see if uaac failed because we've already done this.
-          # Post flight roles need to be idempotent.
-          # uaac will exit with 1 if it has nothing to do and display a message
-          # that says: Nothing to update. Use -i for interactive update.
-          if [ "$?" != "0" ]; then
-            if [[ ! "$uaac_message" == *"Nothing to update"* ]]; then
-              echo $uaac_message
-              exit 1
-            fi
-          fi
-        uaac secret set '<%= client_id %>' --secret '<%= client_data['secret'] %>'
-    <% end %>
-
-    # Re-enable after dealing with uaac client update
-    set -o errexit
-fi
+            %>
+    fi
+    status "- Client <%= client_id %> set secret"
+    uaac --zone "${UAA_ZONE_ID}" secret set '<%= client_id %>' --secret '<%= client_data['secret'] %>'
+    status "- Client <%= client_id %> /done"
+<% end %>
 
-status "Waiting for UAA zone to be available ..."
-retry 240 30s curl ${SILENT} --connect-timeout 5 --fail <%= curl_ssl %> --header 'Accept: application/json' '<%= p("hcf.uaa.internal-url") %>/info'
+status "Waiting for UAA to be availabile at %s..." '<%= p("uaa.url") %>'
```

Comment:
Rebase has restored pre-PR #978 code here. Including the typo. Now hunting for more places where this happened.